pkg/reconciler/pipelinerun/resources/pipelinerunresolution.go

```diff
@@ -227,7 +227,11 @@ func (t *ResolvedPipelineRunTask) parentTasksSkip(facts *PipelineRunFacts) bool
 	stateMap := facts.State.ToMap()
 	node := facts.TasksGraph.Nodes[t.PipelineTask.Name]
 	for _, p := range node.Prev {
-		if stateMap[p.Task.HashKey()].Skip(facts) {
+		parentTask := stateMap[p.Task.HashKey()]
+		if parentTask.Skip(facts) {
+			if facts.ScopeWhenExpressionsToTask && parentTask.PipelineTask.WhenExpressions != nil {
+				continue
```

Comment:
this feels like the opposite of what i would expect - but i might be misunderstanding the purpose of this function! id think that if ScopeWhenExpressionsToTask is true, then it doesnt matter that the parent was skipped?

that for the parent being skipped to matter, ScopeWhenExpressionsToTask would need to be false - or there would need to be a resource dependency?

Reply:
sorry for coming late to the discussion here, I agree with @bobcatfish, if the flag is set to `true`, no need to check if a parent was skipped or not. 

I do not understand this statement:

> if no, skip the current task because this parent task was skipped

A task can be skipped because of the `when` expressions or a task result resolution fails. From the `when` expression scoping perspective, irrespective of one parent skipped vs a couple of parents skipped, the `when` expressions scoped to that parent/s.

Reply:
When i look at the function now, it makes sense to me - not sure if I was seeing something else before that was fixed or if changing the name of the function just cleared it all up for me but all my concerns are addressed!